File: python/56a_1303A_ErasingZeroes.py
def solve(s):
    start, cnt = -1, 0
    for i in range(1, len(s)):
        if s[i] == '0' and s[i-1] == '1':
            start = i
        elif s[i] == '1' and s[i-1] == '0':
            if start is not -1:
                cnt += (i - start)
                start = -1
    return cnt

def main():
    ans = []
    t = int(input())
    for _ in range(t):
        s = input().strip()
        ans.append(solve(s))
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()

File: cpp/56a_1303A_ErasingZeroes.cpp
#include<bits/stdc++.h>

int solve(std::string s){
    int start = -1, cnt = 0;
    for(int i = 1; i < s.size(); i++){
        if(s[i] == '0' && s[i - 1] == '1'){
            start = i;
        }
        else if(s[i] == '1' && s[i-1] == '0'){
            if(start != -1){
                cnt += (i - start);
                start = -1;
            }
        }
    }
    return cnt;
}

int main(){
    int t;
    std::string s;
    std::vector<int> ans;
    std::cin>>t;
    for (int i = 0; i < t; i++){
        std::cin>>s;
        ans.push_back(solve(s));
    }
    for (auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}